src/memory_manager.rs

```diff
@@ -569,13 +568,13 @@ pub fn live_bytes_in_last_gc<VM: VMBinding>(mmtk: &MMTK<VM>) -> usize {
 /// Return the starting address of the heap. *Note that currently MMTk uses
 /// a fixed address range as heap.*
 pub fn starting_heap_address() -> Address {
-    HEAP_START
+    VM_LAYOUT_CONSTANTS.heap_start
```

Comment:
This could accidentally get the `lazy_static` initialised. Consider adding an `&MMTK` reference so the binding won't call it before creating MMTk.

Reply:
This breaks mmtk interface. Also I think a more generic and complete solution would be adding a check in `vm_layout()` getter and `set_custom_vm_layout`, to ensure that the setter can never be called after the first call of `vl_layout()`. This can also check and catch other uses of `vm_layout`, not only the ones here.

Reply:
Sounds good to me. Those checks will have to be `debug_assert` if it is in the getter, right? I am okay with it.

Reply:
Yes I put them within `cfg!(debug_assertions)`